### PR TITLE
fix(netlify, vercel): order route rules from most specific + avoid double-rendering root

### PR DESCRIPTION
**Pull request by danielroe**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This orders route rules within vercel + netlify presets so that (particularly glob) rules do not shadow rules below them. In addition, if the default vercel route rule is `/**: { static: true }` we don't also need to output a second, SSR rule.

cc: @antfu - if you clone this and checkout commit c7f8f39972b43a8ddb0b06d558b3b9f3b19f0a45, then run `vitest run -u` the test passes and the inline snapshots are not updated. I've also noticed some other weird issues like a different snapshot within a file being updated than the one it corresponds to (e.g. the next snapshot below).

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/presets/netlify.ts [hunk 1]
```diff
@@ -125,9 +126,10 @@ async function writeHeaders(nitro: Nitro) {
   const headersPath = join(nitro.options.output.publicDir, "_headers");
   let contents = "";
 
-  for (const [path, routeRules] of Object.entries(
-    nitro.options.routeRules
-  ).filter(([_, routeRules]) => routeRules.headers)) {
+  const rules = Object.entries(nitro.options.routeRules)
+    .sort((a, b) => b[0].split("/").length - a[0].split("/").length)
```
Comment (pi0) [inline src/presets/netlify.ts:130]: Small issue with sorting `/test/a` is more specific than `/test/*` but if * is before, it remains before and shadows.

File: src/presets/vercel.ts [hunk 2]
```diff
@@ -173,10 +173,12 @@ function generateBuildConfig(nitro: Nitro) {
           src: key.replace(/^(.*)\/\*\*/, "(?<url>$1/.*)"),
           dest: generateEndpoint(key) + "?url=$url",
         })),
-      {
-        src: "/(.*)",
-        dest: "/__nitro",
-      },
+      ...(!nitro.options.routeRules['/**']?.cache || !(nitro.options.routeRules['/**'].cache.swr || nitro.options.routeRules['/**']?.cache.static)
```
Comment (pi0) [inline src/presets/vercel.ts:176]: Can you please explain it better in line above? Do we need to also handle other route rules for catch-all such as headers?

File: src/presets/netlify.ts (hunk 1 above)
Comment (danielroe) [reply src/presets/netlify.ts]: nice catch - thank you ❤️

**Review by pi0 — APPROVED** [review]

**Comment by pi0** [comment]
Thanks for the updates. About sorting, it should be same as radix3 structure while hoisting placeholders in regex, might be less accurate i think we can merge it to unblock downstream issue for nuxt.com :)

**Pull request merged**